src/main/java/com/dreamsportslabs/guardian/dao/PasswordlessDao.java

```diff
@@ -46,7 +48,36 @@ public void deletePasswordlessModel(String state, String tenantId) {
     redisClient.rxSend(Request.cmd(Command.DEL).arg(getCacheKey(tenantId, state))).subscribe();
   }
 
+  public Single<Integer> incrementGlobalResendCount(
+      String tenantId, String userIdentifier, Integer ttlSeconds) {
+    String redisKey = getResendCountKey(tenantId, userIdentifier);
+
+    return redisClient
+        .rxSend(Request.cmd(Command.INCR).arg(redisKey))
+        .map(response -> response.toInteger())
+        .flatMap(
+            count -> {
+              if (count == 1) {
+                return redisClient
+                    .rxSend(
+                        Request.cmd(Command.EXPIRE).arg(redisKey).arg(String.valueOf(ttlSeconds)))
+                    .map(__ -> count);
+              }
+              return Maybe.just(count);
+            })
+        .toSingle();
+  }
+
+  public Completable deleteGlobalResendCount(String tenantId, String userIdentifier) {
+    String redisKey = getResendCountKey(tenantId, userIdentifier);
+    return redisClient.rxSend(Request.cmd(Command.DEL).arg(redisKey)).ignoreElement();
+  }
+
   private String getCacheKey(String tenantId, String state) {
     return CACHE_KEY_STATE + "_" + tenantId + "_" + state;
   }
+
+  private String getResendCountKey(String tenantId, String userIdentifier) {
+    return CACHE_KEY_OTP_RESEND_COUNT + ":" + tenantId + ":" + userIdentifier;
```

Comment:
```suggestion
    return CACHE_KEY_OTP_RESEND_COUNT + "_" + tenantId + "_" + userIdentifier;
```

Use a consistent delimiter